### PR TITLE
Use Picker component instead deprecated ClaySelect

### PR DESCRIPTION
**Pull request by JLucasM**
LRAC-13568: [Jira Issue](https://issues.liferay.com/browse/LRAC-13568)

### CODE REVIEW

**Comment by fmarcio** [comment]
This PR needs to be reviewed carefully so please dont forward it for now

**Comment by fmarcio** [comment]
@JLucasM could you please solve the conflicts? Thanks!

File: modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/segment/segment-editor/dynamic/inputs/components/DateFilterConjunctionInput.tsx [hunk 1]
```diff
@@ -50,8 +50,8 @@ const DateFilterConjunctionInput: React.FC<IDateFilterConjunctionInputProps> = (
 		getInitialConjunction(conjunctionCriterion)
 	);
 
-	const handleConjunctionChange = event => {
-		const {value} = event.target;
+	const handleConjunctionChange = newValue => {
+		const {value} = newValue;
```
Comment (interaminense) [inline modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/segment/segment-editor/dynamic/inputs/components/DateFilterConjunctionInput.tsx:54]: Remove this line.

File: modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/segment/segment-editor/dynamic/inputs/components/TimePeriodInput.tsx [hunk 2]
```diff
@@ -10,8 +10,8 @@ interface ITimePeriodInputProps extends ISegmentEditorInputBase {
 
 export default class TimePeriodInput extends React.Component<ITimePeriodInputProps> {
 	@autobind
-	handleTimePeriodChange(event) {
-		const {value} = event.target;
+	handleTimePeriodChange(newValue) {
+		const {value} = newValue;
```
Comment (interaminense) [inline modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/segment/segment-editor/dynamic/inputs/components/TimePeriodInput.tsx:14]: Remove this line.

File: modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/segment/segment-editor/dynamic/inputs/components/attribute-conjunction-input/ValueInput.tsx [hunk 3]
```diff
@@ -45,12 +45,15 @@ const ValueInput: React.FC<IValueInputProps> = ({
...
 					className='boolean-input'
 					data-testid='attribute-value-boolean-input'
+					items={BOOLEAN_OPTIONS.map(option =>
+						createOption(option, dataType)
+					)}
 					onBlur={handleAttributeValueBlur}
-					onChange={event => {
-						const {value} = event.target;
+					onSelectionChange={newValue => {
+						const {value} = newValue;
```
Comment (interaminense) [inline modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/segment/segment-editor/dynamic/inputs/components/attribute-conjunction-input/ValueInput.tsx:56]: Remove this line.

File: modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/segment/segment-editor/dynamic/inputs/components/attribute-conjunction-input/index.tsx [hunk 4]
```diff
@@ -75,8 +75,8 @@ const AttributeFilterConjunctionInput: React.FC<IAttributeFilterConjunctionInput
 		return id;
 	};
 
-	const handleAttributeChange = event => {
-		const {value} = event.target;
+	const handleAttributeChange = newValue => {
+		const {value} = newValue;
```
Comment (interaminense) [inline modules/dxp/apps/osb/osb-faro/osb-faro-web/src/main/js/segment/segment-editor/dynamic/inputs/components/attribute-conjunction-input/index.tsx:79]: Remove this line.

**Review by interaminense — CHANGES_REQUESTED** [review]
Reviewed. Could you take a look at my comments?

**Comment by JLucasM** [comment]
Thaks @interaminense I made the changes that you request.

**Review by interaminense — APPROVED** [review]
LGTM!

**Comment by JLucasM** [comment]
ci:test:sf

**Comment by JLucasM** [comment]
ci:test:relevant

**Review by fmarcio — APPROVED** [review]
LGTM! :rocket:

**Comment by JLucasM** [comment]
ci:test:sf

**Comment by interaminense** [comment]
ci:test:relevant

**Comment by interaminense** [comment]
ci:forward:force

**Pull request closed without merging**